packages/legend-query-builder/src/stores/QueryBuilderState.ts

```diff
@@ -339,13 +340,31 @@ export abstract class QueryBuilderState implements CommandRegistrar {
       this.executionContextState.mapping,
       'Query required mapping to update',
     );
-    queryExeContext.mapping = PackageableElementExplicitReference.create(
-      this.executionContextState.mapping,
-    );
-    queryExeContext.runtime = runtimeValue.packageableRuntime;
+    queryExeContext.mapping = this.executionContextState.mapping.path;
+    queryExeContext.runtime = runtimeValue.packageableRuntime.value.path;
     return queryExeContext;
   }
 
+  async propagateExecutionContextChange(
```

Comment:
would it make more sense to make this an abstract method instead of an extension ?

Reply:
Other child classes e.g. ServiceQueryBuilderState will have their own implementation, the reason why I have to add an extension is that for DataspaceQueryBuilderState, I need to get access to QueryEditorStore and DataspaceQueryBuilderState at the same time, this will introduce an circular dependency in @finos/legend-extension-dsl-data-space